test/e2e/tests/import-tokens.spec.js

```diff
@@ -1,41 +1,72 @@
 const { strict: assert } = require('assert');
 const {
+  defaultGanacheOptions,
   withFixtures,
-  convertToHexValue,
-  regularDelayMs,
   unlockWallet,
+  largeDelayMs,
 } = require('../helpers');
 const FixtureBuilder = require('../fixture-builder');
 
 describe('Import flow', function () {
-  const ganacheOptions = {
-    accounts: [
-      {
-        secretKey:
-          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
-        balance: convertToHexValue(25000000000000000000),
-      },
-    ],
-  };
-
+  async function mockPriceFetch(mockServer) {
+    return [
+      await mockServer
+        .forGet(
+          'https://price-api.metafi.codefi.network/v2/chains/1/spot-prices',
+        )
+        .withQuery({
+          tokenAddresses:
+            '0x06af07097c9eeb7fd685c692751d5c66db49c215,0x514910771af9ca656af840dff83e8264ecf986ca,0x7d4b8cce0591c9044a22ee543533b72e976e36c3',
+          vsCurrency: 'ETH',
+        })
+        .thenCallback(() => {
+          return {
+            statusCode: 200,
+            json: {
+              '0x06af07097c9eeb7fd685c692751d5c66db49c215': {
+                eth: 0.0002,
+              },
+              '0x514910771af9ca656af840dff83e8264ecf986ca': {
+                eth: 0.0003,
+              },
+              '0x7d4b8cce0591c9044a22ee543533b72e976e36c3': {
+                eth: 0.0001,
+              },
+            },
+          };
+        }),
+    ];
+  }
   it('allows importing multiple tokens from search', async function () {
     await withFixtures(
       {
         fixtures: new FixtureBuilder().build(),
-        ganacheOptions,
+        ganacheOptions: defaultGanacheOptions,
         title: this.test.fullTitle(),
+        testSpecificMock: mockPriceFetch,
       },
       async ({ driver }) => {
         await unlockWallet(driver);
 
         // Token list is only on mainnet
         await driver.clickElement('[data-testid="network-display"]');
+        const networkSelectionModal = await driver.findVisibleElement(
+          '.mm-modal',
+        );
+
         await driver.clickElement({ text: 'Ethereum Mainnet', tag: 'p' });
 
         // Wait for network to change and token list to load from state
-        await driver.delay(regularDelayMs);
+        await networkSelectionModal.waitForElementState('hidden');
+        // until we fixed the modal closing before network is switched, this is the temperary solution to wait for network is mainnet
+        await driver.delay(largeDelayMs);
+        const networkDisplayLabel = await driver.findElement(
+          '[data-testid="network-display"]',
+        );
+        assert.equal(await networkDisplayLabel.getText(), 'Ethereum Mainnet');
```

Comment:
The text for this label might change in between these two lines. To verify that an element has certain text content, you should build the expected text into the locator to prevent a race condition.

Reply:
Making this change should let us get rid of that 1 second delay as well